From: Maxim Patlasov <MPatlasov@parallels.com>
Date: Wed, 2 Oct 2013 21:38:54 +0400
Subject: fuse: writepages: protect secondary requests from fuse file release
Git-commit: ce128de6260f86a990ed44a697f26d0859684f28
Patch-mainline: v3.13-rc1
References: FATE#317677

All async fuse requests must be supplied with extra reference to a fuse
file.  This is necessary to ensure that the fuse file is not released until
all in-flight requests are completed.  Fuse secondary writeback requests
must obey this rule as well.

Signed-off-by: Maxim Patlasov <MPatlasov@parallels.com>
Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
---
 fs/fuse/file.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -1487,6 +1487,7 @@ static void fuse_writepage_end(struct fu
 		struct fuse_req *next = req->misc.write.next;
 		req->misc.write.next = next->misc.write.next;
 		next->misc.write.next = NULL;
+		next->ff = fuse_file_get(req->ff);
 		list_add(&next->writepages_entry, &fi->writepages);
 
 		/*
